appinventor/components/src/com/google/appinventor/components/runtime/Sharing.java

```diff
@@ -33,9 +34,12 @@
  *
  * The file path can be taken directly from other components such as the
  * [`Camera`](media.html#Camera) or the [`ImagePicker`](media.html#ImagePicker), but can also be
- * specified directly to read from storage. Be aware that different devices treat storage
- * differently, so a few things to try if, for instance, you have a file called `arrow.gif` in the
- * folder `Appinventor/assets`, would be:
+ * specified directly to read from storage. The default behaviour is to share files from the private
+ * data directory associated with your app. If the file path starts with a slash (`/`), then the file
```

Comment:
Technically these aren't private directories. I would say they are the app-specific directory. The private directory is something different and resides under /data rather than /storage | /sdcard. Prior to Android Q, the QUtil will resolve to /sdcard or equivalent, so isn't even private at all.

Reply:
I just realized the distinction. Will revert the changes made in this PR so far, and implement functionality to allow sharing of app-private files in `/data`.

Using [FileProvider](https://developer.android.com/reference/androidx/core/content/FileProvider)'s [getUriForFile()](https://developer.android.com/reference/androidx/core/content/FileProvider#getUriForFile(android.content.Context,%20java.lang.String,%20java.io.File)) to generate a content URI to a file with temporary access seems to be the way to go.

Reply:
My bad, I understand the distinction now. Have addressed this in my latest commit, where I've made it possible to share files from the app private directory in `/data` instead of the app-specific directory.